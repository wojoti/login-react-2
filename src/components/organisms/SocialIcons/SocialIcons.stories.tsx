/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import SocialIcons from "./SocialIcons";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Organisms/SocialIcons",
  component: SocialIcons,
} as ComponentMeta<typeof SocialIcons>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof SocialIcons> = (args) => (
  <SocialIcons {...args} />
);

export const Default = Template.bind({});
Default.args = {};
