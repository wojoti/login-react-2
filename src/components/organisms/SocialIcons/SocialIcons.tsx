import Icon, { IconType } from "@components/atoms/Icon/Icon";
import Row from "@components/atoms/Row/Row";
import IconButton from "@components/molecules/IconButton/IconButton";
import { useRef } from "react";

export interface SocialIconsProps {
  onIconClick: (icon: string) => void;
  testId?: string;
}

function SocialIcons({ onIconClick, testId }: SocialIconsProps) {
  const googleRef = useRef<HTMLButtonElement>(null);
  const facebookRef = useRef<HTMLButtonElement>(null);
  const linkedinRef = useRef<HTMLButtonElement>(null);
  const onClickG = () => {
    onIconClick("google");
  };
  const onClickF = () => {
    onIconClick("facebook");
  };
  const onClickL = () => {
    onIconClick("linkedin");
  };
  return (
    <Row justify="center" testId={testId}>
      <IconButton
        color="#DE5246"
        ref={googleRef}
        onClick={onClickG}
        testId="iconbutton-g"
      >
        <Icon
          src={IconType.google}
          width={13}
          height={13}
          alt="google"
          testId="icon-g"
        />
      </IconButton>
      <IconButton
        color="#1877f2"
        ref={facebookRef}
        onClick={onClickF}
        testId="iconbutton-fb"
      >
        <Icon
          src={IconType.facebook}
          width={13}
          height={13}
          alt="facebook"
          testId="icon-fb"
        />
      </IconButton>
      <IconButton
        color="#0077B5"
        ref={linkedinRef}
        onClick={onClickL}
        testId="iconbutton-l"
      >
        <Icon
          src={IconType.linkedin}
          width={13}
          height={13}
          alt="linkedin"
          testId="icon-l"
        />
      </IconButton>
    </Row>
  );
}

export default SocialIcons;
