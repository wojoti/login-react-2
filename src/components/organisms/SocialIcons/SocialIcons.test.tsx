import { fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import SocialIcons, { SocialIconsProps } from "./SocialIcons";

const props: SocialIconsProps = {
  onIconClick: () => {},
  testId: "test-socialicons-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<SocialIcons {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render signupform", () => {
  render(<SocialIcons {...props} />);
  const signupformElement = screen.getByTestId("test-socialicons-id");
  expect(signupformElement).toBeInTheDocument();
});

test("should handle clicks", () => {
  const handleClick = jest.fn();
  render(<SocialIcons {...props} onIconClick={handleClick} />);
  const signupformElement = screen.getByTestId("test-socialicons-id");
  expect(signupformElement).toBeInTheDocument();

  const googleElement = screen.getByTestId("iconbutton-g");
  expect(googleElement).toBeInTheDocument();
  const facebookElement = screen.getByTestId("iconbutton-fb");
  expect(facebookElement).toBeInTheDocument();
  const linkedinElement = screen.getByTestId("iconbutton-l");
  expect(linkedinElement).toBeInTheDocument();

  fireEvent.click(googleElement);
  expect(handleClick).toHaveBeenCalledTimes(1);
  expect(handleClick).lastCalledWith("google");
  fireEvent.click(facebookElement);
  expect(handleClick).toHaveBeenCalledTimes(2);
  expect(handleClick).lastCalledWith("facebook");
  fireEvent.click(linkedinElement);
  expect(handleClick).toHaveBeenCalledTimes(3);
  expect(handleClick).lastCalledWith("linkedin");
});
