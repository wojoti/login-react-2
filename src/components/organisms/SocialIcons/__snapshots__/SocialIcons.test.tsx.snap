// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c2 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 13px;
  height: 13px;
}

.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c1 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: #DE5246;
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
}

.c3 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: #1877f2;
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
}

.c4 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: #0077B5;
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
}

<div
  className="c0"
  data-testid="test-socialicons-id"
>
  <button
    className="c1"
    data-testid="iconbutton-g"
    onClick={[Function]}
  >
    <img
      alt="google"
      className="c2"
      data-testid="icon-g"
      height={13}
      src="./img/google.svg"
      width={13}
    />
  </button>
  <button
    className="c3"
    data-testid="iconbutton-fb"
    onClick={[Function]}
  >
    <img
      alt="facebook"
      className="c2"
      data-testid="icon-fb"
      height={13}
      src="./img/facebook.svg"
      width={13}
    />
  </button>
  <button
    className="c4"
    data-testid="iconbutton-l"
    onClick={[Function]}
  >
    <img
      alt="linkedin"
      className="c2"
      data-testid="icon-l"
      height={13}
      src="./img/linkedin.svg"
      width={13}
    />
  </button>
</div>
`;
